src/ioctls/device.rs

```diff
@@ -80,6 +80,104 @@ impl DeviceFd {
         }
         Ok(())
     }
+
+    /// Gets a specified piece of device configuration and/or state.
+    ///
+    /// See the documentation for `KVM_GET_DEVICE_ATTR`.
+    /// # Arguments
+    ///
+    /// * `device_attr` - The device attribute to be get.
+    ///                   Note: This argument serves as both input and output.
+    ///                   When calling this function, the user should explicitly provide
+    ///                   valid values for the `group` and the `attr` field of the
+    ///                   `kvm_device_attr` structure, and a valid userspace address
+    ///                   (i.e. the `addr` field) to access the returned device attribute
+    ///                   data.
+    ///
+    /// # Returns
+    ///
+    /// * Returns the last occured `errno` wrapped in an `Err`.
+    ///
+    /// * `device_attr` - The `addr` field of the `device_attr` structure will point to
+    ///                   the device attribute data.
+    ///
+    /// # Examples
+    /// ```rust
+    /// # extern crate kvm_ioctls;
+    /// # extern crate kvm_bindings;
+    /// # use kvm_ioctls::Kvm;
+    ///
+    /// let kvm = Kvm::new().unwrap();
+    /// let vm = kvm.create_vm().unwrap();
+    ///
+    /// #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+    /// {
+    ///     use kvm_bindings::{
+    ///         KVM_DEV_VFIO_GROUP, KVM_DEV_VFIO_GROUP_ADD,
+    ///         kvm_device_type_KVM_DEV_TYPE_VFIO,
+    ///     };
+    ///
+    ///     // Create a VFIO device.
+    ///     let mut vfio_device = kvm_bindings::kvm_create_device {
+    ///         type_: kvm_device_type_KVM_DEV_TYPE_VFIO,
+    ///         fd: 0,
+    ///         flags: 0,
+    ///     };
+    ///
+    ///     let device_fd = vm
+    ///         .create_device(&mut vfio_device)
+    ///         .expect("Cannot create KVM VFIO device");
+    ///
+    ///     let mut dist_attr_mut = kvm_bindings::kvm_device_attr {
+    ///         group: KVM_DEV_VFIO_GROUP,
+    ///         attr: u64::from(KVM_DEV_VFIO_GROUP_ADD),
+    ///         addr: 0x0,
+    ///         flags: 0,
+    ///     };
+    ///
+    ///     // On x86_64, getting device attributes from `KVM_DEV_VFIO_GROUP`
```

Comment:
In this case we shouldn't have an example with it.
Can you please remove it so it is less verbose?

The point of the examples is to show people how to use the function. It seems like on x86_64 you don't necessarily need it.

Reply:
Done